src/main/ruby/core/io.rb

```diff
@@ -504,8 +504,8 @@ def self.copy_stream(from, to, max_length=nil, offset=nil)
     StreamCopier.new(from, to, max_length, offset).run
   end
 
-  def self.foreach(name, separator=undefined, limit=undefined, options=undefined)
-    return to_enum(:foreach, name, separator, limit, options) unless block_given?
+  def self.foreach(name, separator=undefined, limit=undefined, options=undefined, &block)
+    return to_enum(:foreach, name, separator, limit, options) if block.nil?
```

Comment:
`unless block` seems nicer to me.
`block_given?` is also fine regarding performance FWIW.

Reply:
I can switch it back. But the `if block.nil?` required fewer nodes and produced smaller code.

Reply:
That seem strange to me.
`block_given?` is an AST-inlined method just reading the args and checking if there is a block in it.

`unless block` ought to be less nodes than `if block.nil?` at least.